pkg/registry/minion/rest.go

```diff
@@ -143,15 +143,15 @@ func ResourceLocation(getter ResourceGetter, connection client.ConnectionInfoGet
 	node := nodeObj.(*api.Node)
 	host := node.Name // TODO: use node's IP, don't expect the name to resolve.
 
-	if portReq != "" {
```

Comment:
ah, so whiltelist these https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/master/ports/ports.go? 
we can also filter the endpoint (i.e see if it's in one of the debug endpoints of the kubelet) and pass a different transport in the proxy, or find some way to definitevely pipe the kubelet port to the minion resource (hand waving) because intuition tells me the kubelet port is overrideable.